tests/jobs/test_local_task_job.py

```diff
@@ -55,45 +54,35 @@
 TEST_DAG_FOLDER = os.environ['AIRFLOW__CORE__DAGS_FOLDER']
 
 
-class TestLocalTaskJob(unittest.TestCase):
-    def setUp(self):
-        db.clear_db_dags()
-        db.clear_db_jobs()
-        db.clear_db_runs()
-        db.clear_db_task_fail()
-        patcher = patch('airflow.jobs.base_job.sleep')
-        self.addCleanup(patcher.stop)
-        self.mock_base_job_sleep = patcher.start()
-
-    def tearDown(self) -> None:
+class TestLocalTaskJob:
+    @pytest.fixture(autouse=True)
+    def clean_db_and_set_instance_attrs(self):
         db.clear_db_dags()
         db.clear_db_jobs()
         db.clear_db_runs()
         db.clear_db_task_fail()
```

Comment:
I want a class-level fixture to clear database *after* all the tests run; this prevents side effect from leaking into other tests, which has been problematic in a lot of tests.

Reply:
Cool. I think it is fixed now?

Reply:
So there are three points we need to clean db

1. Before any tests in the class run
2. Between each test in the class
3. After all tests in the class run

Previously you’re missing 3, but now you’re missing 1 and 2 (I think). We need them all.

Reply:
I do not fully understand the pytest fixtures but from what I have read so far, if you don't specify the scope, then the default is function scope which means it's created and cleared at every function run? considering that we are using it on the class. 

I think that if it's not doing 1 & 2 right now that many of the tests on the class would have failed or what do you think?

Reply:
Hi @uranusjr, can you take a look again

Reply:
Yes but you only use the `clear_db` on the class. I’m not sure tbh what that actually does (pytest documentation is quite sparse on this), but I assume it either applies the fixture to be run for every test in the class, or once for that class. Which means you wouldn’t be running the fixture for all of the cases listed above.

Reply:
Just tried, it seems that `usefixtures()` on a class is the same as marking `usefixtures()` on every test in the class. So we are missing a class-level teardown after all tests on the tests are run.

Reply:
Thanks @uranusjr, after much search on this, It seems the fixture is destroyed at the end of each test in the class leaving nothing else to destroy. I experimented by running the tests in /jobs/ and there was no side effect. Another option is using autouse=True and not adding on the class, then it would be used on every test in the module and destroyed at the end of each one.
I think we are on the right path since it runs and destroys on every test leaving nothing behind. I may be missing something though.
https://docs.pytest.org/en/6.2.x/fixture.html#fixture-scopes

Reply:
The fixture is defined like this

```python
@pytest.fixture
def clear_db():
    db.clear_db_dags()
    db.clear_db_jobs()
    db.clear_db_runs()
    db.clear_db_task_fail()
    yield
```

The db clear calls are done in the fixture’s *setup* phase (before yield), so after the last test runs, the fixture itself is destroyed, but that destroy phase does not clear db. So I think we need one more fixture:

```
@pytest.fixture(scope="class")
def clear_db_class():
    yield
    db.clear_db_dags()
    db.clear_db_jobs()
    db.clear_db_runs()
    db.clear_db_task_fail()
```

to handle that?

Reply:
Ok. I will add that